lib/index.js

```diff
@@ -0,0 +1,89 @@
+/*!
+ * Copyright (c) 2017 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const bedrock = require('bedrock');
+const brLedger = require('bedrock-ledger');
+const equihashSigs = require('equihash-signature');
+const validate = require('bedrock-validation').validate;
+const BedrockError = bedrock.util.BedrockError;
+
+require('./config');
+
+bedrock.events.on('bedrock.start', () => brLedger.use('EquihashValidator2017', {
+  type: 'validator',
+  api: api
+}));
+
+const api = {};
+// NOTE: only exported for tests
+module.exports = api;
+
+api.mustValidateEvent = (event, validatorConfig, options, callback) => {
+  if(typeof options === 'function') {
+    callback = options;
+    options = {};
+  }
+  api.validateConfiguration(validatorConfig, err => {
+    if(err) {
+      return callback(err);
+    }
+    if(!validatorConfig.eventFilter.some(f =>
+      f.type === 'EventTypeFilter' && f.eventType.includes(event.type))) {
+      return callback(null, false);
+    }
+    callback(null, true);
+  });
+};
+
+api.validateConfiguration = (validatorConfig, callback) =>
+  validate('ledger-validator-equihash-config', validatorConfig, callback);
+
+api.validateEvent = (event, validatorConfig, callback) => {
+  if(!validatorConfig.eventFilter.some(f =>
```

Comment:
Actually, I think we might want to renegotiate that contract since we overhauled the entire validator system.

The currently implemented _validateEvent helper: https://github.com/digitalbazaar/bedrock-ledger/blob/cm3/lib/index.js#L674 does not use the algorithm/contract you described.

It seems to me that we can keep things very clean by doing it as currently implemented.  Which is that consensus algorithms are NOT responsible for making any determinations about which validators should be run.  THAT is the responsibility of the validators themselves by way of the `mustValidate` API.